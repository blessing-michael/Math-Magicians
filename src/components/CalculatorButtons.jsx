import React, { useState } from 'react';
import propTypes from 'prop-types';

const CalculatorButtons = ({ handleClickedButton }) => {
  const [activeButton, setActiveButton] = useState(null);
  const handleClick = (buttonName) => {
    setActiveButton(buttonName);
    handleClickedButton(buttonName);
  };

  const buttons = [
    { id: 'ac', label: 'AC' },
    { id: 'plus-minus', label: '+/-' },
    { id: 'percent', label: '%' },
    { id: 'divide', label: 'รท' },
    { id: 'seven', label: '7' },
    { id: 'eight', label: '8' },
    { id: 'nine', label: '9' },
    { id: 'multiply', label: 'x' },
    { id: 'four', label: '4' },
    { id: 'five', label: '5' },
    { id: 'six', label: '6' },
    { id: 'minus', label: '-' },
    { id: 'one', label: '1' },
    { id: 'two', label: '2' },
    { id: 'three', label: '3' },
    { id: 'plus', label: '+' },
    { id: 'zero', label: '0', class: 'span-2' },
    { id: 'decimal', label: '.' },
    { id: 'equals', label: '=' },
  ];

  return (
    <div className="all-btns">
      {buttons.map((button) => (
        <button
          key={button.id}
          type="button"
          className={`grid-item ${button.class}`}
          onClick={() => handleClick(button.label)}
          style={activeButton === button.label ? { backgroundColor: 'purple' } : {}}
        >
          {button.label}
        </button>
      ))}
    </div>
  );
};

CalculatorButtons.propTypes = {
  handleClickedButton: propTypes.func.isRequired,
};
export default CalculatorButtons;
