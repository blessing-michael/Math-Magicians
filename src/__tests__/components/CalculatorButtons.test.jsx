import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import CalculatorButtons from '../../components/CalculatorButtons';

test('renders CalculatorButtons component Correctly', () => {
  const tree = renderer.create(<CalculatorButtons handleClickedButton={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('handles button click correctly', () => {
  const handleClickedButton = jest.fn();
  const { getByText } = render(<CalculatorButtons handleClickedButton={handleClickedButton} />);

  const button = getByText('AC'); // Simulate clicking the 'AC' button
  fireEvent.click(button);

  expect(handleClickedButton).toHaveBeenCalledWith('AC'); // Assert that the handleClickedButton function was called with the expected button label
});
