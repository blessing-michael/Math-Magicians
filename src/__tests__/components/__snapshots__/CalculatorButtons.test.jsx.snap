// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CalculatorButtons component Correctly 1`] = `
<div
  className="all-btns"
>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    AC
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    +/-
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    %
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    รท
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    7
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    8
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    9
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    x
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    4
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    5
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    6
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    -
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    1
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    2
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    3
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    +
  </button>
  <button
    className="grid-item span-2"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    0
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    .
  </button>
  <button
    className="grid-item undefined"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    =
  </button>
</div>
`;
